### ISSUE TITLE
Rule to assign followup to an email

### ISSUE THREAD

**Issue by eros419** [tomolimo/mailanalyzer#32]
good morning.

I have a mailbox set up for incidents
The mail is in CC.
we need a rule that compares the title regardless of the presence of the tag [GLPI]. If the rule is a ticket bearing the same title as the email with the same users (email To and CC - user and observer for GLPI), then a follow-up message is created.
This happens in Outlook

**Comment by tomolimo** [comment]
Hello @eros419 
Could you give a detailed example?
Thank you,
Tomolimo

**Comment by eros419** [comment]
![error receptor](https://user-images.githubusercontent.com/57196725/71358532-148f8d00-2568-11ea-9642-29feea37e8e4.JPG)
Attached image with error in the mail receiver
I use GLPI v9.4.4
As an example, every time "Reply to all" or "Reply" is used, an error is generated as it appears in the image, when importing said email from the mail receiver, a new follow-up is generated, and it shows all the Previous conversations when answered by "Reply", "Reply to all" or "Forward", when the last email answered should be displayed.
Sorry for my bad English

**Comment by tomolimo** [comment]
Hello
Could you please provide a step by step explanation, so that I can try to reproduce the issue? 
Could you please provide screen copies in English? 
Thank you 
Regards 
Tomolimo

**Comment by eros419** [comment]
Hello
Sorry for the late reply

1) An email is sent to the email recipient GLPI
2) The applicant receives a confirmation of generated ticket automatically
3) Users are generally answered from the Outlook mail system with a CC copy to the mail of the GLPI mail receiver
Example:
![image](https://user-images.githubusercontent.com/57196725/78261798-11e0f400-74d6-11ea-8a24-1be8eca5e2bf.png)
4) In this process, the automatic action of recovering emails stops, so it must be "forced" from the panel of email recipients.
![image](https://user-images.githubusercontent.com/57196725/78260534-726f3180-74d4-11ea-9248-7c90e39efa49.png)
![image](https://user-images.githubusercontent.com/57196725/78261226-5a4be200-74d5-11ea-872a-deaa57b43c30.png)
5) An error occurs when forcing the reception of emails, but the email is still observed from the ticket tracking panel
![image](https://user-images.githubusercontent.com/57196725/78261328-7c456480-74d5-11ea-848e-2106f87a8526.png)
5) When emails are retrieved, the tracking is updated with the last email generated + previous emails, when you should only bring the last email generated
![image](https://user-images.githubusercontent.com/57196725/78261647-dcd4a180-74d5-11ea-9418-d256b1f26088.png)

 I hope you can understand this explanation and tell me if I perform any procedure wrong
Thanks

**Comment by tomolimo** [comment]
Hello @eros419 
I think I understand your issue.
To rephrase this to be sure that I understood:
1.  User A sends an email E1 to GLPI and in carbon copy to user B
1. In GLPI a ticket is created (T1),
1. User B does a ReplyToAll, and then the new email (E2) is sent to User A and to GLPI
1. Without MailAnalyzer plugin, GLPI will create a new ticket... Not good as there is already one ticket for this subject.
1. With MailAnalyzer plugin, GLPI will add to the previously created ticket T1 a new followup.

Your problem is that MailAnalyzer will not clear from the followup the E1 content? 

thank you,
Regards,
Tomolimo

**Comment by eros419** [comment]
Hello @tomolimo 

It is as you describe it
The problem of not deleting the content of previous emails is that an extensive record is generated and redundancy is generated in the information offered by GLPI
In addition I also have doubts about what is the error when wanting to import new emails that correspond to the same ticket, since the recovery process is not done automatically and must be forced

**Comment by tomolimo** [comment]
Hello @eros419 

1. About the extensive content: the plugin is not designed to clean this. It can be set as an enhancement for a future release. Could create a new issue for that enhancement request?
1. About the error when manually retrieving email from the receiver form: this is not a bug, this is because the plugin will prevent ticket creation when it adds an incoming email to an existing ticket. Than GLPI will "see" this as an error.
1. About the fact that automatic email retrieving is stopped: I cannot say without more information... Could you provide error logs?

Thank you,
Regards,
Tomolimo

**Comment by tomolimo** [comment]
Hello
As there is no feedback provided for the third point, I close the issue. Feel free to reopen when you'll have feedback to provide.
Thank you
Regards,
Tomolimo
